packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts

```diff
@@ -390,9 +390,11 @@ export class OdspSummaryUploadManager {
                         reusedBlobs += result.reusedBlobs;
                         blobs += result.blobs;
                     } else {
-                        // Ideally we should not come here as we should have found it in cache. But in order to successfully upload the summary
```

Comment:
We should not even try to get the summary tree from the cache is `allowHandleExpansion` is false. Only if its true, we should try to get it and log if its not available.

Reply:
I think getting it from cache is fine, as long as cache contains referenced flag.
I believe the issue is that it does not round trip, i.e. TreesLatest does not contain it (but we are fine with incremental summaries). If so, should we push more pressure on SPO folks to round trip it to make whole system work?

Reply:
I mean if allowHandleExpansion is false, then we don't need to get the tree from cache irrespective of whether we have this bug or not.
Yep, the issue is that the tree sent back by the server does not have the unreferenced flag. If we had that, this would work just fine. If we are planning to use cache in this fashion (after your blob aggregation changes as well), then we should push them to send the unreferenced flag back.

Reply:
I think even without this cache requirement, it would be a good idea to get the unreferenced flag. So, we should push them for it.

Reply:
I think we can keep this getting from cache thing as it is not a big op. Otherwise code would become ugly as we would have 2 if-else rather than 1.